src/services/authorization-finder.js

```diff
@@ -36,7 +37,18 @@ function AuthorizationFinder(
       })
       .catch((error) => {
         logger.error('Authorization error: ', error);
-        throw new Error();
+        let errorMessageToForward;
+        switch (error.message) {
+          case errorMessages.SERVER_TRANSACTION.SECRET_AND_RENDERINGID_INCONSISTENT:
+            errorMessageToForward = error.message;
+            break;
+          case errorMessages.SERVER_TRANSACTION.SECRET_NOT_FOUND:
+            errorMessageToForward = 'Cannot retrieve the project you\'re trying to unlock. '
+              + 'Please check that you\'re using the right environment secret regarding your project and environment.';
+            break;
+          default: errorMessageToForward = undefined;
+        }
+        throw new Error(errorMessageToForward);
```

Comment:
I'm really not found of this, you're guessing the error code from the message, that's not robust at all. What if we handle this differently ?

Reply:
Just for the record, we talked about it and it seems that it would be a substantial refact to make it better here.
A future work session on it would be a good idea!